File: download/src/main/java/com/huison/download/utils/FileUtil.java
package com.huison.download.utils;

import java.io.File;

/**
 * Created by huisonma on 2019/5/15.
 */
public class FileUtil {

    public static boolean deleteFile(String path) {
        return deleteFile(new File(path));
    }

    public static boolean deleteFile(File file) {
        if (file == null) {
            return false;
        }
        if (!file.exists()) {
            return true;
        }
        if (file.isFile()) {
            return file.delete();
        }
        return false;
    }

    public static boolean deleteDir(String path) {
        File dir = new File(path);
        if (!dir.exists()) {
            return true;
        }
        if (dir.isDirectory()) {
            File[] list = dir.listFiles();
            boolean result = true;
            for (File child : list) {
                if (result) {
                    result = deleteDir(child.getAbsolutePath());
                } else {
                    deleteFile(child);
                }
            }
            return result;
        } else {
            return deleteFile(dir);
        }
    }

    public static boolean isFileExists(String path) {
        return isFileExists(new File(path));
    }

    public static boolean isFileExists(File file) {
        if (file == null) {
            return false;
        }
        return file.exists();
    }

    public static File createDir(String dirPath) {
        File file = new File(dirPath);
        if (!file.exists()) {
            file.mkdirs();
        }
        return file;
    }

    public static long getFileLength(String path) {
        File file = new File(path);
        if (file.exists()) {
            return file.length();
        }
        return 0;
    }

}

File: ktdownload/src/main/java/com/huison/kotlin/download/utils/FileUtil.kt
package com.huison.kotlin.download.utils

import java.io.File

/**
 * Created by huisonma 2019/9/3.
 */
class FileUtil {

    companion object {

        @JvmStatic
        fun deleteFile(file: File): Boolean {
            if (!file.exists()) {
                return true
            }
            return file.isFile && file.delete()
        }

        @JvmStatic
        fun deleteFile(path: String): Boolean {
            return deleteFile(File(path))
        }

        @JvmStatic
        fun deleteDir(path: String): Boolean {
            val dir = File(path)
            if (!dir.exists()) {
                return true
            }
            var result = true
            if (dir.isDirectory) {
                for (child: File in dir.listFiles()) {
                    if (result) {
                        result = deleteDir(child.absolutePath)
                    } else {
                        deleteFile(child)
                    }
                }
            } else {
                result = deleteFile(path)
            }
            return result
        }

        @JvmStatic
        fun isFileExists(path: String): Boolean {
            return isFileExists(File(path))
        }

        @JvmStatic
        fun isFileExists(file: File): Boolean {
            return file.exists()
        }

        @JvmStatic
        fun createDir(dirPath: String): File {
            val dir = File(dirPath)
            if (!dir.exists()) {
                dir.mkdirs()
            }
            return dir
        }
    }
}